cds_migrator_kit/videos/weblecture_migration/load/load.py

```diff
@@ -26,10 +47,83 @@ def __init__(
     def _prepare(self, entry):
         """Prepare the record."""
         pass
+    
+    def create_publish_single_video_record(self, entry):
+        """Create and publish project and video for single video record."""
+        # Get transformed metadata
+        metadata = entry.get("record", {}).get("json", {}).get("metadata")
+        # Get transformed media files
+        media_files = entry.get("record", {}).get("json", {}).get("media_files")
+        
+        # Create project
+        # TODO `type` will be changed
+        project_metadata = {"category": "CERN", "type": "VIDEO"}
+        project_deposit = create_project(project_metadata)
+        
+        # Create video
+        video_deposit, master_object = create_video(project_deposit, metadata, media_files["master_video"])
+
+        # Get the deposit_id and bucket_id
+        video_deposit_id = video_deposit["_deposit"]["id"]
+        bucket_id = video_deposit["_buckets"]["deposit"]
+        
+        # Create flow and payload
+        flow, payload = create_flow(master_object, video_deposit_id, submitter_id)
+        
+        # Create tags for the master video file
+        init_object_version(flow, has_remote_file_to_download=None)
+        
+        # Extract metadata
+        extract_metadata(payload)
+        
+        # Copy frames
+        frame_paths = media_files["frames"]
+        copy_frames(payload=payload, frame_paths=frame_paths)
+        
+        # Copy subformats
+        subformat_paths = media_files["subformats"]
+        copy_subformats(payload=payload, subformat_paths=subformat_paths)
+
+        # Load additional files
+        additional_files = media_files["additional_files"]
+        copy_additional_files(str(bucket_id), additional_files)
+        
+        # Flow and Tasks modifications need to be persisted
+        db.session.commit()
```

Comment:
what happens if one of the procedures above fails? is there any rollback foreseen?

Reply:
Decided to tackle this in a separate issue